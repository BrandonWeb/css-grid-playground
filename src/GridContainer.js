import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  ${props => props.gridContainerStyle}
`;

class GridContainer extends React.Component {
  render() {
    const childrenWithProps = React.Children.map(this.props.children,
     (child) => React.cloneElement(child, {
       autoHide: this.props.autoHideItemStyle,
       gridItemColor: this.props.gridItemColor
     })
    );
    return (
      <Container
        gridContainerStyle={this.props.gridContainerStyle}>
        {childrenWithProps}
      </Container>
    );
  }

}

export default GridContainer;
