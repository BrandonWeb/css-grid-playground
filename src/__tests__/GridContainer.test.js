import React from 'react';
import renderer from 'react-test-renderer';
import GridContainer from '../GridContainer';

it('renders two grid items correctly', () => {
  const Component = (
    <GridContainer gridContainerStyle="display: grid; grid-template-columns: 1fr 1fr;">
      <div>1</div>
      <div>2</div>
    </GridContainer>
  );
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders four grid items correctly', () => {
  const Component = (
    <GridContainer gridContainerStyle="display: grid; grid-template-columns: 1fr 1fr;">
      <div>1</div>
      <div>2</div>
      <div>3</div>
      <div>4</div>
    </GridContainer>
  );
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
});
