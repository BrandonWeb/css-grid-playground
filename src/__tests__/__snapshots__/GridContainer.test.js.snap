exports[`test renders four grid items correctly 1`] = `
<div
  className="fhHjoh">
  <div>
    1
  </div>
  <div>
    2
  </div>
  <div>
    3
  </div>
  <div>
    4
  </div>
</div>
`;

exports[`test renders two grid items correctly 1`] = `
<div
  className="fhHjoh">
  <div>
    1
  </div>
  <div>
    2
  </div>
</div>
`;
